lib/assets/javascripts/cartodb/table/tableview.js

```diff
@@ -311,6 +312,12 @@
         }
       },
 
+      _dismissSQLHeader: function (e) {
+        this.killEvent(e);
+        this.$('thead').find('.sqlview').remove();
+        this.$('thead > tr').css('height', '');
```

Comment:
I assume you've probably gone through this but just want to double-check that there's nothing in the existing `clearView` pipeline that could be adapted to perform dismissal duties as well.